scripts/format

```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-CLANG_FORMAT=clang-format
+CLANG_FORMAT=clang-format-11
```

Comment:
@dtebbs have you tested this on MacOS? Shouldn't that break? (based on https://formulae.brew.sh/formula/clang-format, `clang-format-<version>` does not seem to be a thing there).
If so, maybe we can keep `CLANG_FORMAT=clang-format` and add a check on the clang-format version below for the script to exit "nicely" if the version isn't the right one, i.e. 11 here.

Reply:
I was planning to address this in the next iteration, but yes `clang-format-11` doesn't exst on mac.  It's unfortunate we can't explicitly name the version we want like this, given the differeces in behaviour.  But to complicate matters, I think more recent versions do a better job of backwards compatibility (I'm using 13 without any issues).  Checking version strings might therefore be complex.  It's a shame, but it might be easiest to just keep `clang-format`.

Reply:
Ok. I think it's fine to stick to `clang-format` here as well. We can skip the version check in the script, but we probably need to extend the README to specify what version is assumed.

One thing though: by keeping `clang-format` as binary here, we need to be careful to not systematically include the modified `format` script whenever we manually change it's value to match our local binary. That'd just be annoying. So this asks for some discipline and best practices (which are important anyway) like: don't always `git add -u` and even less `git add -A`. Instead add the files you strictly want to commit etc etc.

That being said, the caller of the script can also define an alias for `clang-format` (to point to the right version), or - we can simply pass the value `CLANG_FORMAT=clang-format-11` as an env var or input to the script: `./format clang-format-11` (and, I guess `./format clang-format@11` on MacOS?). This is more verbose, but at least that would make it easier to identify version mismatch between local envs and the CI.

Reply:
Removed version number